main.sh

```diff
@@ -91,12 +92,27 @@ trap cleanup_before_exit EXIT
 
 # Translate usage string -> getopts arguments, and set $arg_<flag> defaults
 while read line; do
-  opt="$(echo "${line}" |awk '{print $1}' |sed -e 's#^-##')"
+  # fetch single character version of option sting
+  opt="$(echo "${line}" |awk '{print $1}' |sed -e 's#^-\(.\)#\1#')"
+
+  # fetch long version if present
+  long_opt="$(echo "${line}" |awk 'match($2,"^--.*",a){print substr(a[0],3)}')"
```

Comment:
I think it would be cool to use one method for long/short opt parsing. E.g. go `awk` only for both?